CMakeLists.txt

```diff
@@ -441,14 +441,25 @@ if (WIN32)
 endif (WIN32)
 
 if (UNIX AND NOT PIONEER_INSTALL_INPLACE)
-	set(PIONEER_DESKTOP_FILE ${CMAKE_BINARY_DIR}/metadata/net.pioneerspacesim.Pioneer.desktop)
-	configure_file(metadata/net.pioneerspacesim.Pioneer.desktop.cmakein ${PIONEER_DESKTOP_FILE} @ONLY)
-	install(FILES ${PIONEER_DESKTOP_FILE}
-		DESTINATION ${CMAKE_INSTALL_DATADIR}/applications
-	)
-	install(FILES metadata/net.pioneerspacesim.Pioneer.appdata.xml
-		DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata
-	)
+    if(APPIMAGE_BUILD)
+        set(PIONEER_EXECUTABLE init_pioneer.sh)
+        set(PIONEER_SH_FILE ${CMAKE_BINARY_DIR}/init_pioneer.sh)
+        configure_file(init_pioneer.sh.cmakein ${PIONEER_SH_FILE} @ONLY)
+        install(FILES ${PIONEER_SH_FILE}
+            DESTINATION ${CMAKE_INSTALL_BINDIR}
+            PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_EXECUTE WORLD_EXECUTE
+        )
+    else(APPIMAGE_BUILD)
+        set(PIONEER_EXECUTABLE ${CMAKE_INSTALL_FULL_BINDIR}/pioneer)
+    endif(APPIMAGE_BUILD)
+    set(PIONEER_DESKTOP_FILE ${CMAKE_BINARY_DIR}/metadata/net.pioneerspacesim.Pioneer.desktop)
+    configure_file(metadata/net.pioneerspacesim.Pioneer.desktop.cmakein ${PIONEER_DESKTOP_FILE} @ONLY)
+    install(FILES ${PIONEER_DESKTOP_FILE}
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/applications
+    )
+    install(FILES metadata/net.pioneerspacesim.Pioneer.appdata.xml
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata
+    )
```

Comment:
Please move the `if (APPIMAGE_BUILD)` block (until the endif) right here (after the appdata install). This will shrink the diff even more.